### ISSUE TITLE
WCDT/medbook_UCSF_WXS.20160707.52.input.txt

### ISSUE THREAD

**Issue by melissacline** [BD2KGenomics/core-operations#52]
The input files are to be uploaded to the inbox cgl-inbox-su2c-ucsf

**Comment by melissacline** [comment]
As of 1:30 PM, Paul reported that the tumors had uploaded to S3, and the normals were just beginning to upload.

**Comment by Jeltje** [comment]
7pm, 146 files have been uploaded but not all appear to be paired. Run cannot be started before the weekend.

**Comment by melissacline** [comment]
5 PM Monday: Jeltje is waiting on Paul for a list that pairs the filenames for tumor and normal respectively (there is no single naming convention) and confirmation that the upload is complete.  Some files still seem to be unpaired, so we're suspecting that the upload is not complete yet.

**Comment by melissacline** [comment]
6:30 PM Monday: Paul reports that they are still waiting on a batch from the sequencing center.  He's provided a spreadsheet showing which files to pair together for analysis.  49 pairs are potentially ready to go, out of something like 120.   We're awaiting instructions on whether to run the 49 or to wait .

**Comment by melissacline** [comment]
3:30 PM Thursday: Paul still reports that the data is still uploading.  Changing the status from Ready to Backlog.

**Comment by melissacline** [comment]
11:45 AM Friday: the samples are still uploading.  In preliminary work, Jeltje found three BAMs to be truncated, and asked Paul to re-upload them.
DTB-046-BL-T-DNA-combined_S1.bam
DTB-055-Pro-T-DNA-combined_S1.bam
DTB-059-BL-T-DNA-combined_S1.bam

**Comment by briandoconnor** [comment]
Jeltje suggests we run on what we have by end of today.

**Comment by melissacline** [comment]
Paul reports that everything has finished uploading.  
Just awaiting the tumor/normal pairing

**Comment by Jeltje** [comment]
Tumor/normal sample pairs received, transfering and re-encrypting files

**Comment by adurbin2** [comment]
Adding a comment (test)

**Comment by briandoconnor** [comment]
@Jeltje reports BAM files are truncated!  She's following up with this now.

**Comment by Jeltje** [comment]
The following files have all been uploaded in the last few days and are all truncated. Note that it inclues the three files I reported earlier (I did use the newly uploaded BAM files and checked that they are indeed larger)
DTB-046-BL-T-DNA-combined_S1.bam
DTB-055-Pro-T-DNA-combined_S1.bam
DTB-059-BL-T-DNA-combined_S1.bam
DTB-098-Pro-T-DNA-HSmerge_S1.bam
DTB-102-Pro-T-DNA-HSmerge_S1.bam
DTB-118-BL-T-DNA-HSmerge_S1.bam
DTB-119-Pro-T-DNA-HSmerge_S1.bam
DTB-119-N-DNA.bam
DTB-121-BL-T-DNA-HSmerge_S1.bam
DTB-124-BL-T-DNA-HSmerge_S1.bam
DTB-127-BL-T-DNA-HSmerge_S1.bam
DTB-129-Pro-T-DNA-HSmerge_S1.bam

I have not tried all the recent uploads yet. The earlier uploads all seem fine and are being processed.

**Comment by briandoconnor** [comment]
@melissacline @Jeltje did one of you ping Paul with this list?

**Comment by melissacline** [comment]
Done

**Comment by briandoconnor** [comment]
@melissacline and @Jeltje can we send what you have right now, Chris is in a tight spot to turn around analysis.  Thanks.

**Comment by Jeltje** [comment]
@briandoconnor I am still working on the batch output (we just converged on a gene set this morning). The individual sample pairs have run, but Chris needs cohort level results. Doing what I can.

Meanwhile, Paul has looked at his bam files and they do look complete - there were network issues during his upload, which probably truncated all the latest bam files.

**Comment by Jeltje** [comment]
Sent gene level results for 85 samples off to Chris Wong.
Still need to upload to S3.

**Comment by Jeltje** [comment]
Somehow the last file batch is not getting transfered correctly from the inbox to our encrypted location. I'm using `s3am upload --exists overwrite` and on S3 the files look like they have the same size. However, when I `s3am download` them to the worker, they come out truncated. Downloading the original from `cgl-inbox-su2c-ucsf` creates a complete file.
I'm now doing a rerun on files downloaded directly from the input directory

**Comment by Jeltje** [comment]
Downloading directly appears to work. Started last batch.

**Comment by melissacline** [comment]
Great news!

**Comment by Jeltje** [comment]
Handed off varscan output for 23 samples to Chris Wong (he wanted those first), now running adtex. Three samples still have issues that I have to look into.

**Comment by Jeltje** [comment]
Finished all but 2 samples:
`DTB-129-Pro-T-DNA-HSmerge_S1.bam` is truncated in the inbox
and `samtools index DTB-187-BL-T-DNA-HSmerge_S1.bam` results in a `core dump`, also on the inbox file. File sizes (identical between inbox and download):
```
22199153617 DTB-129-Pro-T-DNA-HSmerge_S1.bam
21116331096 DTB-187-BL-T-DNA-HSmerge_S1.bam
```
Must ask Paul to check these and re-upload

Sent ADTEx and Varscan output of 106 sample pairs to Chris.

**Comment by Jeltje** [comment]
Paul uploaded the straggler files, but I don't see them.

Moved output to S3 and emailed collaborators with the URLs.
Did *not* set requester-pays, @briandoconnor let me know if this is necessary. The output files are very small.

**Comment by briandoconnor** [comment]
@Jeltje requester pays is not required.  Did the missing files get resolved?  There can be a lag in large file uploads (because Amazon constructs the file from multi-part uploads).

**Comment by Jeltje** [comment]
Files never showed. On Paul's end they register as uploaded, but I don't see them on S3 at all. 
@briandoconnor I seem to remember we were going to change the encryption requirement, and use the `awscli` upload method. Is that something we could try here?

**Comment by briandoconnor** [comment]
Hi @Jeltje.  I think you and Paul could try just having Paul directly upload use the AWS CLI, see https://aws.amazon.com/cli/.  If you do that, just make sure he specifies SSE encryption.  It's in the docs.  And upload to a bucket that only he and you can see, e.g. avoid any bucket that's open to the public for read/write.

Alternatively (or maybe before) why don't you try uploading the two files again using the standard approach but change the name... e.g. add like "v2" in the name.  I'm wondering if there's something strange happening here where the fact that the previous version was overwritten and/or deleted.  If you create a new file perhaps this will sidestep an issue with that key?

**Comment by Jeltje** [comment]
Good ideas, thanks. 
It appears Paul is traveling right now and he seems to be of the opinion we should hold off on those last two files until the rest of the batch comes in. I think it's all being sequenced right now. This should be a new issue, obviously, so then we can close this issue.

**Comment by briandoconnor** [comment]
OK, as long as Paul understands he's on the hook for re-uploading in the future.  Don't want to miss these two!  Thanks @Jeltje, have a good vacation!

**Comment by briandoconnor** [comment]
@Jeltje can this be closed?

**Comment by Jeltje** [comment]
@briandoconnor Not quite, but I'm working on it. Need to do some administration on passed and failed samples.

**Comment by Jeltje** [comment]
Created ` WCDT/medbook_UCSF_WXS.20160707.52.input.txt` with the 106 input pairs correctly listed.

Run was performed using the following docker images:
`jeltje/adtex:3`, built from [ADTEx release v1.0.0](https://github.com/Jeltje/adtex/releases/tag/v1.0.0)
`jeltje/varscan` which was built from [Varscan2 release v1.0.0](https://github.com/Jeltje/varscan2/releases/tag/v1.0.0)

Inputs (besides tumor and control bam files):
https://s3-us-west-2.amazonaws.com/varscan-hg19-input/centromeres.bed
https://s3-us-west-2.amazonaws.com/varscan-hg19-input/SeqCapTargets.bed
http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit (converted to hg19.fa)

Outputs (CNV files) and lists with concatenated per-gene scores:
[issue52_urls.txt](https://github.com/BD2KGenomics/core-operations/files/481393/issue52_urls.txt)
